### PR TITLE
Make directory separators platform agnostic

### PR DESCRIPTION
**Pull request by TheCakeIsNaOH**
This does not get everything, specifically, I did not touch the tests, or the automatic checksumming section. 

For the automatic checksumming, I am kind of in favor of putting a check for non-windows, telling people to use Get-RemoteChecksum or embed, then throwing. 

Also, `$ENV:Temp` does not exist on Linux, so I switched that to  `[System.IO.Path]::GetTempPath()`.

Part of #234

### CODE REVIEW

**Comment by majkinetor** [comment]
Isn't slash type irrelevant - '/' works on Windows and \ should work in pwsh on linux ? I just tested it and works fine.

>  I did not touch the tests

Tests should work too cross-platform if we go with this; those that are Windwos specific, if any, could be tagged.

> For the automatic checksumming, I am kind of in favor of putting a check for non-windows, telling people to use Get-RemoteChecksum or embed, then throwing.

I was thinking in deprecating automatic checksum but I guess people use that a lot. So yeah, that, is a must. Also I am not sure about choco dependency in general (but lets talk about that on #234.

**Comment by TheCakeIsNaOH** [comment]
> Isn't slash type irrelevant - '/' works on Windows and \ should work in pwsh on linux ? I just tested it and works fine.

I had issues with `NuspecPath` and `get-remotefiles`.

> Tests should work too cross-platform if we go with this; those that are Windows specific, if any, could be tagged.

I haven't been able to get them to work (yet), but that is good to know.

File: AU/Public/Push-Package.ps1 [hunk 1]
```diff
@@ -14,7 +14,7 @@ function Push-Package() {
         [switch] $All
     )
     $api_key =  if (Test-Path api_key) { Get-Content api_key }
-                elseif (Test-Path ..\api_key) { Get-Content ..\api_key }
+                elseif (Test-Path (Join-Path '..' 'api_key')) { Get-Content (Join-Path '..' 'api_key') }
```
Comment (majkinetor) [inline AU/Public/Push-Package.ps1:17]: Mhm.. this quickly became hard to look at...

Comment (TheCakeIsNaOH) [reply AU/Public/Push-Package.ps1:17]: Yes, it really does. 

That could be resolved by making it into a variable.
`$apikeypath = (Join-Path '..' 'api_key')`

Comment (majkinetor) [reply AU/Public/Push-Package.ps1:17]: Just ranting...

**Pull request merged**